skyportal/utils/cache.py

```diff
@@ -63,28 +68,50 @@ def __setitem__(self, name, data):
         if self._max_items == 0:
             return
 
-        fn = self._hash_fn(name)
+        fn = self._hash_filename(name)
         with open(fn, 'wb') as f:
             f.write(data)
 
         log(f"save [{name}] to [{os.path.basename(fn)}]")
 
         self.clean_cache()
 
-    def clean_cache(self):
-        # Remove stale cache files
-        cached_files = [
-            (f.stat().st_mtime, f.absolute()) for f in self._cache_dir.glob('*')
-        ]
-        cached_files = sorted(cached_files, key=lambda x: x[0], reverse=True)
+    def _remove(self, filenames):
+        """Remove given items from the cache.
+
+        Parameters
+        ----------
+        filenames : list of str
+            Files to remove from the cache.
+        """
         # fmt: off
-        for t, f in cached_files[self._max_items:]:
+        for f in filenames:
             try:
                 os.remove(f)
                 log(f'cleanup [{os.path.basename(f)}]')
             except FileNotFoundError:
                 pass
         # fmt: on
 
+    def clean_cache(self):
+        # Remove stale cache files
+        cached_files = [
+            (f.stat().st_mtime, f.absolute()) for f in self._cache_dir.glob('*')
+        ]
+        cached_files = sorted(cached_files, key=lambda x: x[0], reverse=True)
+
+        now = time.time()
+
+        if self._max_age is not None:
+            removed_by_time = [
+                filename
+                for (mtime, filename) in cached_files
+                if (now - mtime) > self._max_age
+            ]
+            self._remove(removed_by_time)
+
+        oldest = cached_files[self._max_items :]
```

Comment:
Please update to make `max_items` optional -- I want to only specify `max_age` w/ no limit on # cache items.

Reply:
Done